Automattic/wp-calypso
# Zendesk Widget: Pass site ID to when submitting user fields

Fixes 8128-gh-Automattic/dotcom-forge

## Proposed Changes

Pass the site ID, when available, as part of `update-user-fields` request. This allows the `Plan_For_Support` class to fall back to the site when the user doesn't have any plans. 

## Why are these changes being made?
To allow paid site admins who don't own sites of their own to access priority support. 

## Testing Instructions

1. Create a free user and invite them to a paid site you own.
4. Refresh the page in the free user's window.
5. Start a chat session. 
6. The plan should be the site's plan. 

## Regression testing
### Import everything flow
1. Go to /setup/import-focused/importerWordpress?siteSlug=YOUR_SITE&from=https%3A%2F%2Ffriendly-clam.jurassic.ninja%2F&option=everything&forceError=1
2. Click Contact support. 
3. Chat should open just fine.

### Domain only flow
1. Go to /setup/domain-transfer/intro.
2. Click "Need help? Contact us". Chat should start without issues.

### Precancelation chat
1. Cancel a paid plan
2. In "Share your feedback" screen, click the top right button, "Need help? Contact us". 
3. Chat should start alright.
